src/Http/Routing/src/DataSourceDependentCache.cs

```diff
@@ -26,6 +26,11 @@ public DataSourceDependentCache(EndpointDataSource dataSource, Func<IReadOnlyLis
                 throw new ArgumentNullException(nameof(dataSource));
             }
 
+            if (initialize == null)
+            {
+                throw new ArgumentNullException(nameof(initialize));
+            }
+
```

Comment:
I noticed this was missing while re-reviewing this code. It's worth adding because this could would nullref eventually during change token refresh.